asv_bench/benchmarks/benchmarks.py

```diff
@@ -497,29 +498,70 @@ class TimeIndexing:
     param_names = ["shape", "indexer_type"]
     params = [
         get_benchmark_shapes("TimeIndexing"),
-        ["scalar", "bool", "slice", "list", "function"],
+        [
+            "bool_array",
+            "bool_series",
+            "scalar",
+            "slice",
+            "continuous_slice",
+            "numpy_array_take_all_values",
+            "python_list_take_10_values",
+            "function",
+        ],
     ]
 
     def setup(self, shape, indexer_type):
         self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
-        if indexer_type == "bool":
-            self.indexer = [False, True] * (shape[0] // 2)
-        elif indexer_type == "scalar":
-            self.indexer = shape[0] // 2
-        elif indexer_type == "slice":
-            self.indexer = slice(0, shape[0], 2)
-        elif indexer_type == "list":
-            self.indexer = [x for x in range(shape[0])]
-        elif indexer_type == "function":
-            self.indexer = lambda df: df.index[::-2]
+        trigger_import(self.df)
+
+        # This Series is a projection of the source frame, it shouldn't be reimported or triggered to execute.
+        series_bool_mask = self.df.iloc[:, 0] > 50
+        # HACK: Triggering `dtypes` meta-data computation in advance, so it won't affect the `loc/iloc` time.
+        series_bool_mask.dtypes
+
+        self.indexer = {
+            "bool_array": np.array([False, True] * (shape[0] // 2)),
+            "bool_series": series_bool_mask,
+            "scalar": shape[0] // 2,
+            "slice": slice(0, shape[0], 2),
+            "continuous_slice": slice(shape[0] // 2),
+            "numpy_array_take_all_values": np.arange(shape[0]),
+            "python_list_take_10_values": list(range(10)),
+            "function": lambda df: df.index[::-2],
+        }[indexer_type]
```

Comment:
The dictionary certainly looks nice, but in this implementation, all objects will be created for each launch, which will increase the benchmark runtime.

Reply:
done, now the dictionary stores getters instead of actual indexers, also moved its definition to the class level.